Sources/Moya/TargetType.swift

```diff
@@ -49,14 +50,26 @@ public enum UploadType {
 public enum DownloadType {
 
     /// Download a file to a destination.
-    case request(DownloadDestination)
+    case destination(DownloadDestination)
+
+    /// Download a file to a destination with extra parameters using the given encoding.
+    case encoded(DownloadDestination, parameters: [String: Any], encoding: ParameterEncoding)
 }
 
 /// Represents an HTTP task.
 public enum Task {
 
-    /// A basic request task.
-    case request
+    /// A requests body set with data.
+    case requestData(Data)
+
+    /// A requests body set with parameters and encoding.
+    case requestEncoded(parameters: [String: Any], encoding: ParameterEncoding)
```

Comment:
I've renamed to `requestParameters`, although I also considered `requestParameterEncoded` (tell me if you like that more, otherwise I'll keep it).